### ISSUE TITLE
history.py error

### ISSUE THREAD

**Issue by itsasimiqbal** [ipython/ipython#8803]
I am getting this error when I run ipython through my openmind account. Previously, I was able to remove it when I used to remove history (history-corrupt.sqlite  and history.sqlite) but not anymore. Any idea?

```python
Traceback (most recent call last):
  File "/cm/shared/openmind/anaconda/2.1.0/bin/ipython", line 6, in <module>
    sys.exit(start_ipython())
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/__init__.py", line 120, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/config/application.py", line 563, in launch_instance
    app.initialize(argv)
  File "<string>", line 2, in initialize
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/config/application.py", line 92, in catch_config_error
    return method(app, *args, **kwargs)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/terminal/ipapp.py", line 332, in initialize
    self.init_shell()
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/terminal/ipapp.py", line 348, in init_shell
    ipython_dir=self.ipython_dir, user_ns=self.user_ns)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/config/configurable.py", line 354, in instance
    inst = cls(*args, **kwargs)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/terminal/interactiveshell.py", line 328, in __init__
    **kwargs
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 465, in __init__
    self.init_history()
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 1521, in init_history
    self.history_manager = HistoryManager(shell=self, parent=self)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 490, in __init__
    **traits)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 184, in __init__
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
  File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 92, in catch_corrupt_db
    self.init_db()
File "<string>", line 2, in init_db
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 85, in catch_corrupt_db
    return f(self, *a, **kw)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/core/history.py", line 209, in init_db
    self.db = sqlite3.connect(self.hist_file, **kwargs)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/utils/traitlets.py", line 371, in __set__
    obj._notify_trait(self.name, old_value, new_value)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/site-packages/IPython/utils/traitlets.py", line 504, in _notify_trait
    argspec = inspect.getargspec(c)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/inspect.py", line 817, in getargspec
    args, varargs, varkw = getargs(func.func_code)
  File "/cm/shared/openmind/anaconda/2.1.0/lib/python2.7/inspect.py", line 800, in getargs
    return Arguments(args, varargs, varkw)
RuntimeError: maximum recursion depth exceeded while calling a Python object
```

**Comment by minrk** [comment]
@itsasimiqbal is your IPytohn directory on an NFS share or other network filesystem?

**Comment by itsasimiqbal** [comment]
@minrk I think I am on NFS share. I'm new to unix actually and just log in by ssh username@openmind from terminal.

**Comment by minrk** [comment]
What happens if you do 

    IPYTHONDIR=/tmp/ipython-`whoami` ipython

?

**Comment by itsasimiqbal** [comment]
It works! Thanks!



-Asim

**Comment by minrk** [comment]
@itsasimiqbal and what if you do 

    IPYTHONDIR=$HOME/tmpipython ipython

?

**Comment by itsasimiqbal** [comment]
Then it doesn't work!



-Asim

**Comment by minrk** [comment]
Then there's something bizarre about your NFS system that doesn't allow creating even one sqlite file on it. I'm not sure if there's a workaround we can do, but if you put the history file on the local scratch filesystem, it should work.

If you set:

    c.HistoryManager.hist_file = '/path/to/somewhere/on/scratch/history.sqlite'

Then it should work.

**Comment by takluyver** [comment]
I think we could be a bit smarter: at present, if it fails to create the database, it just keeps retrying. It could retry once, and then fall back to `:memory:` or a temp file if it fails the second time.

**Comment by gregrtw** [comment]
@takluyver Hey! how can I contribute to IPython and work on this particular bug?

Cheers!

**Comment by takluyver** [comment]
This one was already fixed by #9418, I think - sorry @gregrtw .

Jupyter contributor guides in general are here: http://jupyter.readthedocs.io/en/latest/contributor/content-contributor.html
